add 1.4.13.4 for SZ

disheng222 opened the pull request:
We release a new version for SZ, in which there are some useful functions added.
Details about versions are README of http://github.com/disheng222/SZ
Thanks.

adamjstewart commented on var/spack/repos/builtin/packages/sz/package.py:
```diff
@@ -30,10 +30,13 @@ class Sz(AutotoolsPackage):
 
     homepage = "https://collab.cels.anl.gov/display/ESR/SZ"
-    url      = "https://github.com/disheng222/SZ/archive/v1.4.11.0.tar.gz"
+    url      = "https://github.com/disheng222/SZ/archive/v1.4.13.4.zip"
```
If you change the extension, none of the older versions will download. Can you keep it `.tar.gz` and update the checksums?

adamjstewart commented on var/spack/repos/builtin/packages/sz/package.py:
```diff
@@ -1,12 +1,12 @@
 # LLNL-CODE-647188
 #
-# For details, see https://github.com/spack/spack
+# For details, see https://github.com/llnl/spack
```
Can you undo the changes to the license? `github.com/llnl/spack` is out of date.

adamjstewart commented:
Is this a duplicate of #8372?

disheng222 commented:
Hi, 
I changed the package.py according to your comment. 
Please feel free to let me know more questions/suggestions if any.
Thanks.

disheng222 commented:
Hi Adam,
I just changed the package.py and submitted a new "pull request" based on
your comment.

Please feel free to let me know more questions/suggestions if any.
I submitted a pull request yesterday, but I guess that one was rejected.

Thanks.

Best,
Sheng

2018-06-05 13:41 GMT-05:00 Adam J. Stewart <notifications@github.com>:

> Is this a duplicate of #8372 <https://github.com/spack/spack/pull/8372>?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/spack/spack/pull/8375#issuecomment-394817100>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJW1SWSpeC93CteYJyKpIYzwDQkSmb6Tks5t5tDNgaJpZM4UaKer>
> .
>

adamjstewart commented:
> I submitted a pull request yesterday, but I guess that one was rejected.

It wasn't rejected, it's still open.

adamjstewart commented on var/spack/repos/builtin/packages/sz/package.py:
```diff
@@ -25,15 +25,15 @@
 
 from spack import *
 
-
```
Deleting this line will cause flake8 to fail.

ax3l commented:
@disheng222 when you are adding new `sz` releases, can you please verify the `adios` package still builds? `sz` is constantly changing it's API in patch-level releases (e.g. `1.4.12` to `1.4.13`) which breaks the adios build, e.g. 1.13.1 which is in spack :-)

no problem, but please consider running `spack install adios` after you added a new release.

ax3l commented:
Fixed it in  #8408

Also: @disheng222 congrats on the new SZ release! :)

disheng222 commented:
Hi Axel,
Thanks for the information.
We changed SZ API only once in history and I notified Adios developer about
this. They said they will tune it in their release.
I will contact Adios developer for further information.
Thanks!

Best,
Sheng

On Wed, Jun 6, 2018 at 11:14 AM, Axel Huebl <notifications@github.com>
wrote:

> @disheng222 <https://github.com/disheng222> when you are adding new sz
> releases, can you please verify the adios package still builds? sz is
> constantly changing it's API in patch-level releases (e.g. 1.4.12 to
> 1.4.13) which breaks the adios build, e.g. 1.13.1 which is in spack :-)
>
> no problem, but please consider running spack install adios after you
> added a new release.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spack/spack/pull/8375#issuecomment-395126729>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJW1STmk9J5LikvOH2Zv4Prppfn6LLXpks5t5__3gaJpZM4UaKer>
> .
>

ax3l commented:
@disheng222 thanks for the information!
Huh, confusing - I got compile errors for adios on changes to `1.4.10`, `.11` and `.12`. Any idea what could be causing this?

Thanks,
Axel

disheng222 commented:
Hi Jon,
Axel said there are compile errors for Adios with SZ, probably because of
changed API from 1.4.9 to 1.4.10.
I checked the API. I guess you are using SZ_compress_args function, and it
did have some changes from 1.4.9 and 1.4.10.
1.4.9:
unsigned char *SZ_compress_args(int dataType, void *data, int *outSize, int
errBoundMode, double absErrBound, double relBoundRatio, int r5, int r4, int
r3, int r2, int r1);
1.4.10:
unsigned char* SZ_compress_args(int dataType, void *data, size_t *outSize,
int errBoundMode, double absErrBound,
double relBoundRatio, double pwrBoundRatio, int pwrType, size_t r5, size_t
r4, size_t r3, size_t r2, size_t r1);

In the 1.4.10+, pwrBoundRatio means point-wise relative error bound ratio
and pwrType could be set to 0 for simplicity.
If the users set the error bound mode to be absolute error bound or
value-range based relative error bound, pwrBoundRatio and pwrType will be
ignored.

Could you change the corresponding codes in ADIOS to fix the compile issue?
Thanks in advance!

Best,
Sheng

On Wed, Jun 6, 2018 at 1:05 PM, Axel Huebl <notifications@github.com> wrote:

> @disheng222 <https://github.com/disheng222> thanks for the information!
> Huh, confusing - I got compile errors for adios on changes to 1.4.10, .11
> and .12. Any idea what could be causing this?
>
> Thanks,
> Axel
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spack/spack/pull/8375#issuecomment-395161290>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJW1SRe9z6T5-6p7ecNhlByaC-rR7iq-ks5t6BnXgaJpZM4UaKer>
> .
>

disheng222 commented:
Hi Jong,
I just changed the API again in order to make it easy for you.
I removed the "pwrType" argument from SZ_compress_args(), because this
argument will be deprecated in the future SZ version.
If you used SZ_compress_args() in your code, please change it
correspondingly, as follows. Thanks.
unsigned char* SZ_compress_args(int dataType, void *data, size_t *outSize,
int errBoundMode, double absErrBound,
double relBoundRatio, double pwrBoundRatio, size_t r5, size_t r4, size_t
r3, size_t r2, size_t r1)

If you actually didn't use SZ_compress_args() but SZ_compress() to do the
compression, you don't need to change your code.
Any questions are welcome!

Best,
Sheng

On Wed, Jun 6, 2018 at 1:59 PM, Sheng Di <disheng222@gmail.com> wrote:

> Hi Jon,
> Axel said there are compile errors for Adios with SZ, probably because of
> changed API from 1.4.9 to 1.4.10.
> I checked the API. I guess you are using SZ_compress_args function, and it
> did have some changes from 1.4.9 and 1.4.10.
> 1.4.9:
> unsigned char *SZ_compress_args(int dataType, void *data, int *outSize,
> int errBoundMode, double absErrBound, double relBoundRatio, int r5, int r4,
> int r3, int r2, int r1);
> 1.4.10:
> unsigned char* SZ_compress_args(int dataType, void *data, size_t *outSize,
> int errBoundMode, double absErrBound,
> double relBoundRatio, double pwrBoundRatio, int pwrType, size_t r5, size_t
> r4, size_t r3, size_t r2, size_t r1);
>
> In the 1.4.10+, pwrBoundRatio means point-wise relative error bound ratio
> and pwrType could be set to 0 for simplicity.
> If the users set the error bound mode to be absolute error bound or
> value-range based relative error bound, pwrBoundRatio and pwrType will be
> ignored.
>
> Could you change the corresponding codes in ADIOS to fix the compile
> issue? Thanks in advance!
>
> Best,
> Sheng
>
> On Wed, Jun 6, 2018 at 1:05 PM, Axel Huebl <notifications@github.com>
> wrote:
>
>> @disheng222 <https://github.com/disheng222> thanks for the information!
>> Huh, confusing - I got compile errors for adios on changes to 1.4.10, .11
>> and .12. Any idea what could be causing this?
>>
>> Thanks,
>> Axel
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/spack/spack/pull/8375#issuecomment-395161290>, or mute
>> the thread
>> <https://github.com/notifications/unsubscribe-auth/AJW1SRe9z6T5-6p7ecNhlByaC-rR7iq-ks5t6BnXgaJpZM4UaKer>
>> .
>>
>
>

ax3l commented:
@disheng222 @jychoi-hpc the problem I am currently seeing is that ADIOS 1.13.1 does not compile with SZ 1.4.13.* but only with 1.4.12.*

ADIOS 1.13.0 only works with SZ 1.4.11 (not any later) and ADIOS 1.12.0 only works with SZ 1.4.10 (not any later). Maybe you can figure out the source of this? According to @disheng222 the public API has not changed between the releases.

jychoi-hpc commented:
Just did a quick look-through, ADIOS is not using SZ_compress_args. Instead, it calls the following SZ APIs:
```
SZ_Init
SZ_Init_Params(&sz);
SZ_compress
SZ_decompress
SZ_Finalize
```
One time, I remember there was a change of size from int to size_t one of the parameters. I guess some compilers will make only warning or you will get a hard error.

Regarding the using the most recent version of SZ, I have just checked using ADIOS 1.13.1 with SZ 1.4.13.5 and found that there is some changes in sz_params structure. Basically, a few members are removed (dataEndianType, sysEndianType, etc). I can fix this from the ADIOS side.

ax3l commented:
@disheng222 are the APIs @jychoi-hpc mentions stable or should other SZ routines be used?

disheng222 commented:
The SZ routines are stable. Based on Jong's answers, I guess the reason is
that I deleted some deprecated parameters in a global data structure
instead of API.
I believe Jong already knows how to fix it in Adios.
Best,
Sheng

On Thu, Jun 7, 2018 at 9:13 AM, Axel Huebl <notifications@github.com> wrote:

> @disheng222 <https://github.com/disheng222> are the APIs @jychoi-hpc
> <https://github.com/jychoi-hpc> mentions stable or should other SZ
> routines be used?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spack/spack/pull/8375#issuecomment-395435203>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJW1SZPZLM6DxGYR5bT7xuDtYA48yTKRks5t6TT3gaJpZM4UaKer>
> .
>

ax3l commented:
Alright, thanks @disheng222!
